number_theory/numeric.h

```diff
@@ -43,6 +45,54 @@ std::pair<T, T> exgcd(T a, T b) {
   return std::make_pair(xa, ya);
 }
 
+// Accumulates values according to the binary representation of |binary|.
```

Comment:
I don't think the behaviour of this function is clear enough. Why are we accumulating a function's results on each bit and return it? What is the purpose of this function?

Reply:
I add an example to help understand this function. You can treat it as std::accumulate for binary representations.

Reply:
Popcount is a great example, nice job!